packages/storybook/src/migrations/update-12-5-0/migrate-storybook-6-3.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  formatFiles,
+  getProjects,
+  joinPathFragments,
+  logger,
+  readJson,
+  Tree,
+  updateJson,
+} from '@nrwl/devkit';
+import { checkAndCleanWithSemver } from '@nrwl/workspace/src/utilities/version-utils';
+import { gte, lt } from 'semver';
+
+let needsInstall = false;
+const targetStorybookVersion = '6.3.0';
+
+function getPackageVersion(json, packageName: string) {
+  let packageVersion = json.dependencies[packageName];
+
+  if (!packageVersion) {
+    packageVersion = json.devDependencies[packageName];
+  }
+
+  if (packageVersion) {
+    return checkAndCleanWithSemver(packageName, packageVersion);
+  } else {
+    return null;
+  }
+}
+
+/**
+ * Upgrades to Storybook v6.3 (if currently a 6.x version is installed).
+ * Also for Angular projects makes sure the `@storybook/builder-webpack5` and `@storybook/manager-webpack5`
+ * are installed
+ */
+function upgradeStorybook63(tree: Tree) {
+  updateJson(tree, 'package.json', (json) => {
+    json.dependencies = json.dependencies || {};
+    json.devDependencies = json.devDependencies || {};
+
+    function updatePackageSection(
+      storybookPackageName: string,
+      packageSection: 'dependencies' | 'devDependencies'
+    ) {
+      if (json[packageSection][storybookPackageName]) {
+        const version = checkAndCleanWithSemver(
+          storybookPackageName,
+          json[packageSection][storybookPackageName]
+        );
+        if (gte(version, '6.0.0') && lt(version, targetStorybookVersion)) {
+          json[packageSection][
+            storybookPackageName
+          ] = `~${targetStorybookVersion}`;
+          needsInstall = true;
+        }
+      }
+    }
+
+    const storybookPackages = [
+      '@storybook/angular',
+      '@storybook/react',
+      '@storybook/addon-knobs',
+      '@storybook/addon-controls',
```

Comment:
maybe we could change this to `@storybook/addon-essentials`, to work with the knobs-to-controls?